### ISSUE TITLE
Add ability to access formData when not valid

### ISSUE THREAD

**Issue by michaelryancaputo** [mozilla-services/react-jsonschema-form#307]
### Prerequisites
- [x] I have read the [documentation](https://github.com/mozilla-services/react-jsonschema-form/blob/master/README.md#readme).
### Description

Would like the option to save data if the form is invalid.
### Steps to Reproduce
1. Create an invalid form
2. Submit the form
3. onError is triggered, and doesn't allow us access to formData
#### Expected behavior

Would like to be able to access invalid form data from onError.
#### Actual behavior

The validation warnings are only accessible from onError.
### Version

`0.39.0`

**Comment by elisechant** [comment]
can't you just use the `onSubmit` handler and post form artificially from `formData` args available there?

**Comment by michaelryancaputo** [comment]
@elisechant Possibly, it's been a while since i've looked at this, but based on the documentation (https://github.com/mozilla-services/react-jsonschema-form#form-submission) ```You can pass a function as the onSubmit prop of your Form component to listen to when the form is submitted and its data are valid. It will be passed a result object having a formData attribute, which is the valid form data you're usually after```

**Comment by michaelryancaputo** [comment]
I just tested it, and no the `formData` is not available if the form is not valid.

**Comment by LucianBuzzo** [comment]
Using `onChange` and storing the `formData` value will work well here - take a look at this example https://codesandbox.io/s/14ovp8nwk3
